### PR TITLE
Added tests for Elytron security realms

### PR DESCRIPTION
**Pull request by honza-kasik**
* Code is ready
* Test documentation is missing for now, I will add it if it is required

### CODE REVIEW

**Comment by honza-kasik** [comment]
Added documentations to test cases

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/securityrealm/ElytronJDBCSecurityRealmTestCase.java [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+        public ModelNode getPrincipalQueries() {
+            return listBuilder.build();
+        }
+
+        public Address getAddress() {
+            return address;
+        }
+
+        public static final class Builder {
```
Comment (zebrik) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/securityrealm/ElytronJDBCSecurityRealmTestCase.java]: I don't suggest to use `public` visibility for class (Builder) inside of a `private` class (JDBCRealm)  ;-) 
Moreover both of them are and should be called only from the top level class (ElytronJDBCSecurityRealmTestCase) so there is no reason in making any of them public.

**Review by zebrik — COMMENTED** [review]
LGTM except of one minor complaint about nested class visibility.

**Comment by honza-kasik** [comment]
Thanks! I made Builder `private`.

**Comment by zebrik** [comment]
LGTM. @honza-kasik do you feel it's suitable to do (partial ?) squashing in this case?

**Pull request merged**